alf/layers.py

```diff
@@ -729,13 +730,27 @@ def __init__(self,
         initialization. An auto gain calculation might depend on the activation
         following the causal conv1d layer.
 
+        Note that the main difference of causal conv v.s. standard conv is that
+        each temporal element in the convolutional output is causal w.r.t.
+        the temporal elements from input. For example, for a length L sequence
+        x with the shape of [B, C, L], and y = causal_conv(x), where the shape
+        of y is [B, C', L], by causal we mean y[..., l] only depends on
+        X[..., :l] (i.e. the past), and there is no dependency on X[..., :l]
+        (i.e. future) as in the standard non-causal convolution.
+
+        This can implemented by using an asymmetric padding, which in effect
+        shift the input to the right (future) according to kernel size.
+
         Args:
             in_channels (int): channels of the input
             out_channels (int): channels of the output
             kernel_size (int): size of the kernel
             dilation (int): controls the spacing between the kernel points.
                 Please refer to here for a visual illustration:
                 https://github.com/vdumoulin/conv_arithmetic/blob/master/README.md
+            hide_current (bool): whether to hide the current by shifting the
```

Comment:
If the current info in the first layer is missing, how will the later layers get it?